test/gtest/ucp/test_ucp_mmap.cc

```diff
@@ -314,8 +318,12 @@ void test_ucp_mmap::test_length0(unsigned flags)
     status = ucp_mem_map(sender().ucph(), &params, &memh[1]);
     ASSERT_UCS_OK(status);
 
+    bool expect_rma_offload = is_tl_rdma() ||
+                              ((flags & UCP_MEM_MAP_ALLOCATE) &&
+                               is_tl_shm());
+
     for (i = 0; i < buf_num; i++) {
-        test_rkey_management(&sender(), memh[i], true, false);
+        test_rkey_management(&sender(), memh[i], true, expect_rma_offload);
```

Comment:
i don't understand how the PR fixes test failure: before the change, we passed expect_rma_offload:false to test_rkey_management(), so the EXPECT_NE should not have been called anyway because of ` if (expect_rma_offload && is_dummy) {`

Reply:
i.e. SW RMA proto has to be used in case of dummy protocol in case dummy and even if it is RMA offload

Reply:
@yosefe are you agree with the fix?

Reply:
can't we always pass "false" here for expect_rma_offload?

Reply:
I think it is better to check what we select in case of dummy proto, so also `rand()` not always covers `0` case. So, here we always will cover it

Reply:
so maybe like this:
```
if (expect_rma_offload) {
            EXPECT_EQ(&ucp_rma_basic_proto, rkey->cache.rma_proto);
     } else {
            EXPECT_EQ(&ucp_rma_sw_proto, rkey->cache.rma_proto);
   }
```

and set expect_rma_offload accordingly

Reply:
then we have to check `is_dummy()` for tests that use `rand()` to generate message size and `is_dummy` check will be considered in `expect_rma_offload`.
are you ok with it?